zephyr: Update to zephyr v2.4.0.

MaureenHelm opened the pull request:
Minor changes to the zephyr port to update to zephyr v2.4.0. 

~~This PR is in draft status until the zephyr release is finalized.~~
The PR is now ready to go.

~~@tejlmand will you please have a look at the commit `zephyr: Move system includes to the end of Z_CFLAGS.`~~
A minor zephyr build system bug was fixed and this workaround commit was removed.

MaureenHelm committed:
zephyr: Replace zephyr integer types with C99 types.

Zephyr v2.4.0 stopped using custom integer types in favor of C99 types
instead.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
zephyr: Const-ify struct device instance pointers.

Zephyr v2.4.0 added a const qualifier to usages of struct device to
allow storing device driver instances exclusively in flash and thereby
reduce ram footprint.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
travis: Update zephyr build to v2.4.0.

Updates CI to use the latest zephyr release tag.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm commented:
Updated from zephyr v2.4.0-rc1 to -rc2 and rebased to pick up 5c503de52166f6ebd4ef59b58b130a730782823d

tejlmand commented on ports/zephyr/CMakeLists.txt:
```diff
@@ -18,7 +18,7 @@ add_custom_target(
   COMMAND echo CC="${CMAKE_C_COMPILER}"
   COMMAND echo AR="${CMAKE_AR}"
-  COMMAND echo Z_CFLAGS=${system_includes} ${includes} ${definitions} ${options}
+  COMMAND echo Z_CFLAGS=${includes} ${definitions} ${options} ${system_includes}
```
This should not be needed.
Fixed the issue here: https://github.com/zephyrproject-rtos/zephyr/pull/28584
that could result in an empty `-isystem`.

tejlmand reviewed (commented):
> @tejlmand will you please have a look at the commit zephyr: Move system includes to the end of Z_CFLAGS.

Bug fix PR: zephyrproject-rtos/zephyr#28584

MaureenHelm commented:
Updated from zephyr v2.4.0-rc2 to -rc3

MaureenHelm committed:
zephyr: Update build instructions to v2.4.0.

Updates the zephyr port build instructions to use the latest zephyr
release tag.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm commented:
Updated from zephyr v2.4.0-rc3 to the final v2.4.0 release. This PR is now ready to go

dpgeorge commented:
Thank you!  Rebased and merged in ce49be43b1ac7582edcfec20c56928781a512a7d through 997ec9e8ccaf4f1d8a950ef025e098e6b033c66c
